include/deal2lkit/parsed_preconditioner.h

```diff
@@ -0,0 +1,224 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2015 by the deal2lkit authors
+//
+//    This file is part of the deal2lkit library.
+//
+//    The deal2lkit library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal2lkit distribution.
+//
+//-----------------------------------------------------------
+
+#ifndef _d2k_parsed_preconditioner_h
+#define _d2k_parsed_preconditioner_h
+
+#include <deal2lkit/config.h>
+#include <deal.II/base/config.h>
+#include <deal.II/lac/trilinos_precondition.h>
+#include <deal2lkit/parsed_finite_element.h>
+#include <deal2lkit/parameter_acceptor.h>
+#include <deal2lkit/utilities.h>
+
+using namespace dealii;
+
+
+D2K_NAMESPACE_OPEN
+
+/**
+ * A parsed AMG preconditioner which uses parameter files to choose
+ * between different options. This object is a
+ * TrilinosWrappers::PreconditionAMG which can be called in place of
+ * the preconditioner.
+ */
+class ParsedAMGPreconditioner : public ParameterAcceptor, public TrilinosWrappers::PreconditionAMG
+{
+public:
+  /**
+   * Constructor. Build the preconditioner of a matrix using AMG.
+   */
+  ParsedAMGPreconditioner(const std::string &name = "",
+                          const bool &elliptic = true,
+                          const bool &higher_order_elements = false,
+                          const unsigned int &n_cycles = 1,
+                          const bool &w_cycle = false,
+                          const double &aggregation_threshold = 1e-4,
+                          const std::string &var_const_modes = "none",
+                          const unsigned int &smoother_sweeps = 2,
+                          const unsigned int &smoother_overlap = 0,
+                          const bool &output_details = false,
+                          const std::string &smoother_type = "Chebyshev",
+                          const std::string &coarse_type = "Amesos-KLU"
+                         );
+
+  /**
+   * Declare preconditioner options.
+   */
+  virtual void declare_parameters(ParameterHandler &prm);
+
+  /**
+   * Initialize the preconditioner using @p matrix.
+   */
+  template<int dim, int spacedim=dim, typename Matrix>
+  void initialize_preconditioner(const ParsedFiniteElement<dim, spacedim > &fe,
```

Comment:
If I don't want to use costant modes, I don't need to feed the initialize with both a `FiniteElement` and a `DoFHandler`